tests/upgrade-tests/validation.py

```diff
@@ -86,3 +88,19 @@ def validate_status_messages(model):
         assert response.status_code == 401  # Unauthorized
     units = model.applications['kubernetes-worker'].units
     await asyncio.gather(*(validate_unit(unit) for unit in units))
+
+
+async def validate_e2e_tests(model):
+    ''' Validate that the e2e tests pass.'''
+    masters = model.applications['kubernetes-master']
+    await masters.set_config({'allow-privileged': 'true'})
+    workers = model.applications['kubernetes-worker']
+    await workers.set_config({'allow-privileged': 'true'})
+    if len(workers.units) < 2:
+        await workers.add_unit(1)
+    await wait_for_ready(model)
+
+    e2e_unit = model.applications['kubernetes-e2e'].units[0]
+    action = await e2e_unit.run_action('test')
+    await action.wait()
+    assert action.status == 'completed'
```

Comment:
I've gotten a ton of testing in on this validator over the past week while testing 1.7. Solid :+1: